examples/basics/hw_decode.py

```diff
@@ -0,0 +1,76 @@
+import os
+import time
+
+import av
+import av.datasets
+
+# What accelerator to use.
+# Recommendations:
+#   Windows:
+#       - d3d11va (Direct3D 11)
+#           * available with built-in ffmpeg in PyAV binary wheels, and gives access to
+#             all decoders, but performance may not be as good as vendor native interfaces.
+#       - cuda (NVIDIA NVDEC), qsv (Intel QuickSync)
+#           * may be faster than d3d11va, but requires custom ffmpeg built with those libraries.
+#   Linux (all options require custom FFmpeg):
+#       - vaapi (Intel, AMD)
+#       - cuda (NVIDIA)
+#   Mac:
+#       - videotoolbox
+#           * available with built-in ffmpeg in PyAV binary wheels, and gives access to
+#             all accelerators available on Macs. This is the only option on MacOS.
+
+HW_DEVICE = os.environ["HW_DEVICE"] if "HW_DEVICE" in os.environ else None
+
+if "TEST_FILE_PATH" in os.environ:
+    test_file_path = os.environ["TEST_FILE_PATH"]
+else:
+    test_file_path = av.datasets.curated(
+        "pexels/time-lapse-video-of-night-sky-857195.mp4"
+    )
+
+if HW_DEVICE is None:
+    av.codec.hwaccel.dump_hwdevices()
+    print("Please set HW_DEVICE.")
+    exit()
+
+assert HW_DEVICE in av.codec.hwaccel.hwdevices_available, f"{HW_DEVICE} not available."
```

Comment:
I think the example is a little outdated here:
- there is no `dump_hwdevices` member in `av.codec.hwaccel`
- `av.codec.hwaccel.hwdevices_available` is a `cyfunction`, not a list, set or dict

Reply:
https://github.com/PyAV-Org/PyAV/pull/1689 should fix it.